pudb/debugger.py

```diff
@@ -1103,6 +1123,35 @@ def examine_frame(w, size, key):
 
         self.stack_list.listen("enter", examine_frame)
 
+        def open_file_editor(file_name, line_number):
+            try:
+                filename_edited = self.debugger.open_file_to_edit(file_name,
+                                                                  line_number)
+            except Exception:
+                from pudb.lowlevel import format_exception
+                self.message("Exception happened when trying to edit the file:"
+                             "\n\n%s" % ("".join(format_exception(sys.exc_info()))),
+                    title="File Edit Error")
+                return
+
+            self.message("File is changed, but the execution is continued with"
+                         " the 'old' codebase.\n"
+                         f"Changed file: {filename_edited}\n\n"
+                         "Please quit and restart to see changes",
+                         title="File is changed")
+
+        def open_editor_on_stack_frame(w, size, key):
+            _, pos = self.stack_list._w.get_focus()
+            index = self.translate_ui_stack_index(pos)
+
+            curframe, line_number = self.debugger.stack[index]
+            file_name = curframe.f_code.co_filename
+
+            open_file_editor(file_name, line_number)
+            redraw_screen(w, size, key)
```

Comment:
Why is this necessary? (Instead, you should stop/start the Urwid screen, which may switch the terminal between front and alternate buffers.)

Reply:
We don't need it 🙂  Good catch!

Reply:
DONE